staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto

```diff
@@ -107,6 +107,11 @@ service RuntimeService {
 
     // Status returns the status of the runtime.
     rpc Status(StatusRequest) returns (StatusResponse) {}
+
+    // CheckpointContainer checkpoints a container
+    rpc CheckpointContainer(CheckpointContainerRequest) returns (CheckpointContainerResponse) {}
+    // RestoreContainer restores a container
+    rpc RestoreContainer(RestoreContainerRequest) returns (RestoreContainerResponse) {}
```

Comment:
does this apply to all types of containers ? e.g. init and sidecars flagged as well?

Reply:
> does this apply to all types of containers ? e.g. init and sidecars flagged as well?

In my `drain --checkpoint` implementation (https://github.com/kubernetes/kubernetes/pull/97194) I am ignoring containers in the 'kube-system' namespace. As long as no raw sockets are used CRIU could checkpoint most processes and containers, but it does not seem useful to checkpoint containers in the 'kube-system' namespace for now.

Checkpointing a Pod in CRI-O also does not checkpoint the pause container, only necessary metadata is written to the checkpoint archive and upon restore a new pause container is created which uses the metadata to provide the same environment for the other restored containers.

Reply:
Thanks @adrianreber , i was asking about https://kubernetes.io/docs/concepts/workloads/pods/init-containers/
Also does it make sense for the CRI to know about kubernetes namespace kube-system at the CRI level ? I believe that logic should be built into kubelet and not into CRI

Reply:
>  i was asking about https://kubernetes.io/docs/concepts/workloads/pods/init-containers/

This was briefly discussed in the corresponding KEP and one idea was to not enable checkpointing before init containers have finished running.

> Also does it make sense for the CRI to know about kubernetes namespace kube-system at the CRI level ? I believe that logic should be built into kubelet and not into CRI

Currently it happens in kubectl. Only containers which are not in the 'kube-system' namespace are checkpointed.